### PR TITLE
reconstruct the indexing page

### PR DESCRIPTION
**Pull request by feipenghe**
This PR fixes https://github.com/asyml/forte/issues/643. 

### Description of changes
* reorganize outline to avoid conflicted sections
* add descriptions to finished sections and make sure one hop to the target section
* Fix expired image links

### Possible influences of this PR.
Forte documentation would be much clearer.

### Test Conducted
Visually check the html rendering effect.

### CODE REVIEW

File: docs/index_toc.rst [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+Chapter 8. Chatbot
+==================
+.. toctree::
+   :maxdepth: 2
+
+   toc/chatbot.md
+
+.. rst-class:: page-break
+
+Chapter 9. Task on Other modalities
```
Comment (hunterhector) [inline docs/index_toc.rst:88]:
```suggestion
Chapter 9. Tasks on Other modalities
```

File: docs/index_toc.rst [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+Chapter 6. Text Generation
+====================================
+.. toctree::
+   :maxdepth: 2
+
+   toc/text_generation.md
+
+.. rst-class:: page-break
+
+Chapter 7. Question Answering System
```
Comment (hunterhector) [inline docs/index_toc.rst:70]:
```suggestion
Chapter 7. Question Answering
```

File: docs/toc/audio_processing.md [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+
+
+## Automatic Speech Recognition
+Automatic Speech Recognition (ASR) develops methodologies and technologies that enable the recognition and translation of spoken language into text by computers. Here we using a simple example to show how to build a forte pipeline to perform speech transcription tasks. This example is based on a pretrained wav2vec2 model and you can check out the details [here](https://huggingface.co/facebook/wav2vec2-base-960h).
+## Speaker Segmentation
+Speaker segmentation consists in partitioning a conversation between one or more speakers into speaker turns. It is the process of partitioning an input audio stream into acoustically homogeneous segments according to the speaker identity. A typical speaker segmentation system finds potential speaker change points using the audio characteristics. In this example, the speaker segmentation is backed by a pretrained Hugging Face model where you can find details [here](https://huggingface.co/pyannote/speaker-segmentation).
+
+## Example script
+For full example script, please refer to [this audio example](https://github.com/feipenghe/forte/tree/master/examples/audio)
```
Comment (hunterhector) [inline docs/toc/audio_processing.md:36]: Do you want to update it to the master repo?

Comment (hunterhector) [reply docs/toc/audio_processing.md:36]: no, just a reminder for you to update the link to the master link

**Review by hunterhector — APPROVED** [review]
Very good work, the documentation should be much nicer after this.

**Review by hunterhector — COMMENTED** [review]
By looking at the build here: https://asyml-forte--642.org.readthedocs.build/en/642/, a few small suggestions:

1. Each title appear twice (*overview*, *NLP with Forte*, *Appendices* and *API* all appear twice)
2. The front page is basically with only links. It will be better to make the front page a description. (The text in overview will be a good choice).

**Comment by feipenghe** [comment]
> By looking at the build here: https://asyml-forte--642.org.readthedocs.build/en/642/, a few small suggestions:
> 
> 1. Each title appear twice (_overview_, _NLP with Forte_, _Appendices_ and _API_ all appear twice)
> 2. The front page is basically with only links. It will be better to make the front page a description. (The text in overview will be a good choice).

Duplicate title is unavoidable due to considerations of showing sections correctly on the left bar.
I rewrote `overview.md` to `rst` format and append the content to `index.rst`.

**Review by hunterhector — APPROVED** [review]
The duplicated titles are also gone after you rewrite the overview. thanks

**Pull request merged**